Project.toml

```diff
@@ -4,24 +4,32 @@ authors = ["Sebastijan Dumancic <s.dumancic@tudelft.nl>", "Jaap de Jong <J.deJon
 version = "0.3.1"
 
 [deps]
+CSV = "336ed68f-0bac-5ca0-87d4-7b16caf5d00b"
+Clingo_jll = "e1d4ad0e-dc8b-5d2d-8155-e9a55b531f08"
+DataFrames = "a93c6f00-e57d-5684-b7b6-d8193f3e46c0"
 DataStructures = "864edb3b-99cc-5e75-8d2d-829cb0a9cfe8"
 HerbConstraints = "1fa96474-3206-4513-b4fa-23913f296dfc"
 HerbCore = "2b23ba43-8213-43cb-b5ea-38c12b45bd45"
 HerbGrammar = "4ef9e186-2fe5-4b24-8de7-9f7291f24af7"
 HerbInterpret = "5bbddadd-02c5-4713-84b8-97364418cca7"
 HerbSpecification = "6d54aada-062f-46d8-85cf-a1ceaf058a06"
+JSON = "682c06a0-de6a-54ab-a142-c8b1cf79cde6"
 Logging = "56ddb016-857b-54e1-b83d-db4d58db5568"
 MLStyle = "d8e11817-5142-5d16-987a-aa16d5891078"
 Random = "9a3f8284-a2c9-5f02-9a11-845980a1fd5c"
 StatsBase = "2913bbd2-ae8a-5f71-8c99-4fb6c76f3a91"
 
 [compat]
+CSV = "0.10.15"
+Clingo_jll = "5.7.1"
```

Comment:
We have to load `Clingo` all the time you want to use `HerbSearch`. Is there any way around this?

Reply:
The instantiation time of Clingo_jll seems minimal, as shown in the figure (instantiation time of all packages). If you want the refactoring to be a part of the main functionality (as it is in the PR now) it seems best to me already load the package in, as the pre-compilation probably speeds up the code. 
![image](https://github.com/user-attachments/assets/a6ad9638-8119-405d-935a-f6dcb873a152)
Otherwise, lazy package loading using requires.jl could be a solution ([Github](https://github.com/JuliaPackaging/Requires.jl)).